plugins/logitech-hidpp/fu-logitech-hidpp-device.c

```diff
@@ -1319,10 +1319,13 @@ fu_logitech_hidpp_device_init(FuLogitechHidppDevice *self)
 FuLogitechHidppDevice *
 fu_logitech_hidpp_device_new(FuUdevDevice *parent)
 {
-	return g_object_new(FU_TYPE_HIDPP_DEVICE,
+	FuLogitechHidppDevice *self = NULL;
+	self = g_object_new(FU_TYPE_HIDPP_DEVICE,
 			    "proxy",
 			    parent,
 			    "device-file",
 			    fu_udev_device_get_device_file(parent),
 			    NULL);
+	fu_udev_device_set_io_channel(FU_UDEV_DEVICE(self), fu_udev_device_get_io_channel(parent));
```

Comment:
I'm a bit confused -- can't we open the device in the child? it's the reason we're passing the `fu_udev_device_get_device_file`

Reply:
Current implementation doesn't open the file, so devices connected via Bolt got:
```
failed to send: (null) [Logi M750] has not been opened
```
while sending something to them.
They are created via directly called new/probe/setup calls, see for instance https://github.com/fwupd/fwupd/blob/7afc6b226de74391e6460738bf94c3098683a153/plugins/logitech-hidpp/fu-logitech-hidpp-runtime-bolt.c#L204
And I found that for now we are sending data through https://github.com/fwupd/fwupd/blob/7afc6b226de74391e6460738bf94c3098683a153/plugins/logitech-hidpp/fu-logitech-hidpp-hidpp.c#L62 which is directly calling `fu_udev_device_write()` which fails without io_channel.

Do you think it is better to call fu_device_open() everywhere and use dedicated descriptors for each child?

Reply:
> Do you think it is better to call fu_device_open() everywhere

I think that's a much clearer lifecycle -- and probably a lot less complicated to understand. If it's possible, that is. It'll also play much nicer for emulation too.